src/scenes/App/router.js

```diff
@@ -1,73 +1,70 @@
-import React, { PropTypes } from 'react';
+import React from 'react';
+import PropTypes from 'prop-types';
 import { Switch, Route } from 'react-router-dom';
 
 import asyncComponent from '../../helpers/AsyncFunc';
-import NavBar from '../../components/NavBar/index';
 import { RouterPath } from '../../constants';
 
-export default class AppRouter extends React.Component {
-  static propTypes = {
-    url: PropTypes.string.isRequired,
-    langHandler: PropTypes.func.isRequired,
-  };
+const AppRouter = ({ url, langHandler }) => {
+  // Remove trailing '/' from url so that we can use `${url}/topic` below
+  if (url[url.length - 1] === '/') {
+    url = url.slice(0, url.length - 1); // eslint-disable-line
```

Comment:
this `eslint-disable-line` was put here because eslint doesn't want us reassigning arguments, but in this case, it doesn't really affect anything. See example here https://spin.atomicobject.com/2011/04/10/javascript-don-t-reassign-your-function-arguments/